File: js/ds/Queue.js
class Queue {
  constructor() {
    this.data = [];
    this.size = 0;
  }

  enqueue(value) {
    this.data.push(value);
    this.size++;
  }

  dequeue() {
    this.size--;
    return this.data.shift();
  }

  length() {
    return this.size;
  }

  print() {
    console.log(this.data);
  }

  isEmpty() {
    return this.size === 0;
  }

  getFront() {
    return this.isEmpty() ? null : this.data[0];
  }

  getLast() {
    return this.isEmpty() ? null : this.data[this.size - 1];
  }
}

const queue = new Queue();
queue.enqueue(5);
queue.enqueue(6);
queue.enqueue(7);
console.log(queue.length());
console.log(queue.dequeue());
console.log(queue.dequeue());
console.log(queue.dequeue());
console.log(queue.length());

File: python/ctci/ds/queue.py


class Queue:
    def __init__(self):
        self.items = []

    def is_empty(self):
        return len(self.items) == 0

    def enqueue(self, value):
        self.items.append(value)
        return

    def dequeue(self):
        item_removed = self.items[0]
        print(item_removed)
        self.items.pop(0)
        return


q = Queue()
q.enqueue(1)
q.enqueue(2)
q.enqueue(3)
q.enqueue(4)
q.dequeue()
q.dequeue()
q.dequeue()
